timeless/views.py

```diff
@@ -30,86 +49,148 @@ def delete(self):
         pass
 
 
-class GenericView(views.View):
+class GenericView(views.MethodView):
```

Comment:
@bkuzmic  I believe there should be `views.View`, since `views.MethodView` is made for API calls, but generic is for templates

Reply:
`views.MethodView ` is just to check if the request.method is in the allowed methods, that's all, so I wouldn't change it.

Reply:
@bkuzmic yes, sounds reasonable